tracing-journald/src/lib.rs

```diff
@@ -150,6 +152,34 @@ impl Subscriber {
         self
     }
 
+    /// Adds fields that will get be passed to journald with every log entry.
+    ///
+    /// The input values of this function are interpreted as `(field, value)` pairs.
+    ///
+    /// This can for example be used to configure the syslog facility.
+    /// See [Journal Fields](https://www.freedesktop.org/software/systemd/man/systemd.journal-fields.html)
+    /// and [journalctl](https://www.freedesktop.org/software/systemd/man/journalctl.html)
+    /// for more information.
```

Comment:
i agree with @Ralith's comment in https://github.com/tokio-rs/tracing/pull/2708#issuecomment-1703127097 that we shouldn't try to protect the user from sending duplicates of existing fields, since a solution for that that works with arbitrary span fields could be quite complex. However, I do think that we ought to have the documentation clearly state what the behavior is when field names overlap.

Could we maybe add something like:
```suggestion
    /// Adds fields that will get be passed to journald with every log entry.
    ///
    /// The input values of this function are interpreted as `(field, value)` pairs.
    ///
    /// This can for example be used to configure the syslog facility.
    /// See [Journal Fields](https://www.freedesktop.org/software/systemd/man/systemd.journal-fields.html)
    /// and [journalctl](https://www.freedesktop.org/software/systemd/man/journalctl.html)
    /// for more information.
    ///
    /// Fields specified using this method will be added to the journald
    /// message alongside fields generated from the event's fields, its
    /// metadata, and the span context. If the name of a field provided using
    /// this method is the same as the name of a field generated by the
    /// subscriber, both fields will still be recorded.
``` 

(i'm open to changing the wording)

Reply:
*"both fields will still be recorded."* - what does that mean? Will journald actually record both, or will it discard one of them? Is this even protocol conform?

Reply:
maybe "both fields will be sent to journald", then.

Reply:
if we know what journald's behavior in this case is, i'm happy to include that in the documentation as well, but the most important thing is making sure that what `tracing-journald` will do is documented.

Reply:
Agree. Fixed.